modules/clientBinaries/generateVersions.js

```diff
@@ -0,0 +1,202 @@
+  };
+
+  // windows binaries
+  if (asset.name.includes("win")) {
+    getWinAsset(data, asset, hashes);
+  } // osx binaries
+  else if (asset.name.includes("osx")) {
+    getOSXAsset(data, asset, hashes);
+  } // linux binaries
+  else if (asset.name.includes("linux")) {
+    getLinuxAsset(data, asset, hashes);
+  }
+
+  // add .exe extension for windows binaries
+  var bin = `particld${data.platform === 'win' ? '.exe' : ''}`
+  // return asset only if it is fully compliant
+  return (data.platform && data.arch && data.type ? {
+    platform: data.platform,
+    arch: data.arch,
+    name: asset.name,
+    entry: {
+      download: {
+        url: asset.browser_download_url,
+        type: data.type,
+        sha256: data.sha256,
+        bin: `particl-${version}/bin/${bin}`
+      },
+      bin: bin,
+      commands: {
+        sanity: {
+          args: ["-version"],
+          output: ["Particl Core Daemon", version]
+        }
+      }
+    }
+  } : undefined);
+}
+
+/*
+ * Gets all hashes of current version for a specific platform
+ */
+var getHashesForPlatform = function (platform, path, hashes, promises) {
+  // this promise is resolved when both HTTP calls returned
+  return new Promise((resolve, reject) => {
+
+    got(`${signaturesURL}/${path}/${maintainer}`).then(response => {
+
+      var files = JSON.parse(response.body);
+      for (id in files) {
+        if (!files[id].name.includes("assert.sig")) {
+
+          got(`${files[id].download_url}`).then(response => {
+            hashes[platform] = response.body;
+            resolve(response.body);
+          }).catch(error => reject(error)); /* sig file */
+
+        }
+      }
+
+    }).catch(error => reject(error)); /* folder containing sig files */
+
+  });
+}
+
+/*
+ * Entry point
+ * get Particl latest release files
+ */
+got(`${releasesURL}`).then(response => {
+
+  var release = JSON.parse(response.body)[0];
+  var version = release.tag_name.substring(1);
+  var binaries = [];
+
+  // get gitian repository of hashes
+  got(`${signaturesURL}`).then(response => {
+
+    // prepare JSON object for the output file
+    var json = {
+      clients: {
+        particld: {
+          version: version,
+          platforms: {}
+        }
+      }
+    }
+
+    var versions = JSON.parse(response.body);
+    var hashes = {};
+    var promises = [];
+
+    for (id in versions) {
+      // select folders that match the current version
+      if (versions[id].name.includes(version)) {
+        // extract matching folder's platform
+        var platformIndex = versions[id].name.indexOf("-");
+        var platform = versions[id].name.substring(platformIndex + 1);
+        // wait for hashes to be added to our hashes array
+        promises.push(getHashesForPlatform(platform, versions[id].name, hashes));
+      }
+    }
+
+    // once we have all hashes
+    Promise.all(promises).then(function () {
+      // get asset details for each release entry
+      for (id in release.assets) {
```

Comment:
I'm not sure if this is an array?
arrays should use array.forEach(... => {
...
})